lib/drivers/insuletDriver.js

```diff
@@ -637,6 +696,266 @@ module.exports = function (config) {
     return wizRecords;
   };
 
+  var buildAlarmRecords = function (data, records) {
+    var alarmrecs = findSpecificRecords(data.log_records, [
+      getValueForName(logRecords, 'Remote_Hazard_Alarm'),
+      getValueForName(logRecords, 'Alarm')
+    ]);
+    var pumpAlarms = findPumpAlarmRecords(data.log_records);
+    alarmrecs = alarmrecs.concat(pumpAlarms);
+    var postrecords = [];
+
+    function makeSuspended(anAlarm) {
+      var suspend = cfg.builder.makeDeviceMetaSuspend()
+        .with_time(anAlarm.timestamp || alarm.alarm.timestamp)
+        .with_deviceTime(anAlarm.deviceTime || alarm.alarm.deviceTime)
+        .with_timezoneOffset(anAlarm.timezoneOffset || alarm.alarm.timezoneOffset)
+        .with_reason({suspended: 'automatic'});
+      if (anAlarm.detail) {
+        suspend.set('index', anAlarm.log_index);
+      }
+      return suspend.done();
+    }
+
+    function makeSuspendBasal(anAlarm) {
+      // we don't call .done() on the basal b/c it still needs duration added
+      // which happens in the simulator
+      var basal = cfg.builder.makeSuspendBasal()
+        .with_time(anAlarm.timestamp || alarm.alarm.timestamp)
+        .with_deviceTime(anAlarm.deviceTime || alarm.alarm.deviceTime)
+        .with_timezoneOffset(anAlarm.timezoneOffset || alarm.alarm.timezoneOffset);
+      if (anAlarm.detail) {
+        basal.set('index', anAlarm.log_index);
+      }
+      return basal;
+    }
+
+    for (var a = 0; a < alarmrecs.length; ++a) {
+      var alarm = data.log_records[alarmrecs[a]];
+      var postalarm = null, postsuspend = null, postbasal = null;
+      var alarmValue = null;
+
+      postalarm = cfg.builder.makeDeviceMetaAlarm()
+        .with_time(alarm.timestamp || alarm.alarm.timestamp)
+        .with_deviceTime(alarm.deviceTime || alarm.alarm.deviceTime)
+        .with_timezoneOffset(alarm.timezoneOffset || alarm.alarm.timezoneOffset);
+
+      // handle history-style alarms
+      if (alarm.detail) {
+        alarmValue = alarm.detail.alarm_type;
+        postalarm.set('index', alarm.log_index);
+      }
+      // handle non-history alarms
+      else {
+        // alarm.alarm.alarm is not a typo!
+        if (alarm.alarm.alarm_text != null) {
+          alarmValue = alarm.alarm.alarm;
+        }
+        else {
+          postalarm = null;
+        }
+      }
+      switch (alarmValue) {
+        // History - ALARM
+        case 21:
```

Comment:
Can you please use the constants defined for the alarms?

case ALARM_TYPES.ALARM_KEY.value: